.github/actions/release-extension/action.yml

```diff
@@ -0,0 +1,93 @@
+
+    # Grabs the latest version from the extensions.json file
+    # If an extension hasn't been released yet we default to `0.0.0` so any
+    # version in the manifest will be higher
+    - name: Get lastest version from extension list
+      continue-on-error: true
+      run: |
+        LATEST_VERSION=$(jq -c '.extensions[] | select(.name=="${{ inputs.extension-name }}").latestVersion.version' < extensions.json)
+        LATEST_VERSION=$(semver -c "${LATEST_VERSION:-0.0.0}")
+        echo "LATEST_VERSION=$LATEST_VERSION" >> "$GITHUB_ENV"
+      shell: bash
+
+    # We only want to release if the manifest.json contains a newer semver
+    # version than the latest version in the extensions.json
+    # We compare that here, and echo if a release will occur
+    # This can be helpful when looking at Pull Request action outputs
+    # so it is clear what will happen on a merge to `main`
+    - name: Check if manifest has newer version
+      id: should_release
+      run: |
+        echo "The last released version is '$LATEST_VERSION' and the manifest version is '$MANIFEST_VERSION'"
+        HIGHER_VERSION=$(semver "$MANIFEST_VERSION" "$LATEST_VERSION" | tail -n 1)
+          if [ "$MANIFEST_VERSION" = "$HIGHER_VERSION" ] && [ "$MANIFEST_VERSION" != "$LATEST_VERSION" ]; then
+          echo "Should release: the manifest version is higher than the lastest version"
+          echo "should_release=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "Should not release: the manifest version is not higher than the lastest version"
+          echo "should_release=false" >> "$GITHUB_OUTPUT"
+        fi
+      shell: bash
+
+    # Here we download the packaged extension artifact to release
+    - uses: actions/download-artifact@v4
+      if: github.ref_name == 'main' && steps.should_release.outputs.should_release == 'true'
+      with:
+        name: ${{ inputs.extension-name }}.tar.gz
+
+    # The release tag utilizes both the extension name and semver version
+    # to create a unique tag for the repository
+    - name: Release tag
+      if: github.ref_name == 'main' && steps.should_release.outputs.should_release == 'true'
+      id: release_tag
+      run: |
+        RELEASE_TAG="${{ inputs.extension-name }}@v$MANIFEST_VERSION"
+        echo "RELEASE_TAG=$RELEASE_TAG" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Release
+      if: github.ref_name == 'main' && steps.should_release.outputs.should_release == 'true'
+      run: |
+        gh release create $RELEASE_TAG \
+          --title "${{ inputs.extension-name }} v$MANIFEST_VERSION" \
+          ${{ inputs.extension-name }}.tar.gz
+      shell: bash
+
+    # We fetch the full release data to utilize in the extension list update
+    - name: Get release data
+      if: github.ref_name == 'main' && steps.should_release.outputs.should_release == 'true'
+      run: gh api /repos/${{ github.repository }}/releases/tags/$RELEASE_TAG > release-${{ inputs.extension-name }}.json
```

Comment:
More for my education / curiosity: what's in the `release-....json`?

Reply:
Ah interesting. It might be worth stating (more?) explicitly in the comment "We fetch the full release data from the github API" That _wasn't_ at the top of my head for where we were fetching that from, actually.

Reply:
Updated the comment in https://github.com/posit-dev/connect-extensions/pull/35/commits/e5340888b0e86d4cb5616466e63b45d21d9181ae to mention that the release data is fetched from the GitHub API 👍